src/engine/InputQuery.cpp

```diff
@@ -867,6 +869,100 @@ bool InputQuery::constructReluLayer( NLR::NetworkLevelReasoner *nlr,
     return true;
 }
 
+bool InputQuery::constructLeakyReluLayer( NLR::NetworkLevelReasoner *nlr,
+                                          Map<unsigned, unsigned> &handledVariableToLayer,
+                                          unsigned newLayerIndex )
+{
+    INPUT_QUERY_LOG( "Attempting to construct LeakyReLULayer..." );
+    struct NeuronInformation
+    {
+    public:
+
+        NeuronInformation( unsigned variable, unsigned neuron, unsigned sourceVariable )
+            : _variable( variable )
+            , _neuron( neuron )
+            , _sourceVariable( sourceVariable )
+        {
+        }
+
+        unsigned _variable;
+        unsigned _neuron;
+        unsigned _sourceVariable;
+    };
+
+    List<NeuronInformation> newNeurons;
+
+    // Look for ReLUs where all b variables have already been handled
+    const List<PiecewiseLinearConstraint *> &plConstraints =
+        getPiecewiseLinearConstraints();
+
+    double alpha = 0;
+    for ( const auto &plc : plConstraints )
+    {
+        // Only consider ReLUs
+        if ( plc->getType() != LEAKY_RELU )
+            continue;
+
+        const LeakyReluConstraint *relu = (const LeakyReluConstraint *)plc;
```

Comment:
relu --> leakyRelu

while it's reasonable that a leaky relu is a relu, I suggest to call it a leak relu to avoid confusion in the code

Reply:
fixed